homeassistant/components/telegram_bot/__init__.py

```diff
@@ -296,48 +333,58 @@ def _get_msg_ids(self, msg_data, chat_id):
         return message_id, inline_message_id
 
     def _get_target_chat_ids(self, target):
-        """Validate chat_id targets or return default target (fist defined).
+        """Validate chat_id targets or return default target (first).
 
-        :param target: optional list of strings or ints (['12234'] or [12234])
+        :param target: optional list of integers ([12234, -12345])
         :return list of chat_id targets (integers)
         """
         if target is not None:
-            if isinstance(target, int):
-                if target in self.allowed_chat_ids:
-                    return [target]
-                _LOGGER.warning("BAD TARGET %s, using default: %s",
+            try:
+                chat_ids = [int(t) for t in target
+                            if int(t) in self.allowed_chat_ids]
+                if len(chat_ids) > 0:
+                    return chat_ids
+                _LOGGER.warning("Unallowed targets: %s", target)
+            except (ValueError, TypeError):
+                _LOGGER.warning("Bad target data: %s, using default: %s",
```

Comment:
Shouldn't the warning about default  be included in the above warning instead? The warning here should never happen.